cpp/include/cudf/column/column_view.hpp

```diff
@@ -567,4 +569,27 @@ size_type count_descendants(column_view parent);
  */
 column_view logical_cast(column_view const& input, data_type type);
 
+/**
+ * @brief Zero-copy cast between types with the same underlying representation.
+ *
+ * This is similar to `reinterpret_cast` or `bit_cast` in that it gives a view of the same raw bits
+ * as a different type. Unlike `reinterpret_cast` however, this cast is only allowed on types that
+ * have the same width and underlying representation. For example, the way timestamp types are laid
+ * out in memory is equivalent to an integer representing a duration since a fixed epoch; logically
+ * casting to the same integer type (INT32 for days, INT64 for others) results in a raw view of the
+ * duration count. However, an INT32 column cannot be logically cast to INT64 as the sizes differ,
+ * nor can an INT32 columm be logically cast to a FLOAT32 since what the bits represent differs.
+ *
+ * The validity of the conversion can be checked with `cudf::is_logically_castable()`. For other
+ * conversions between fixed-width types which require a copy, see `cudf::cast()`.
```

Comment:
I realize that this just copy and pasted from above, but is it really that we "require a copy" - it is more that is just isn't "logically castable"

Reply:
I've removed this redundant sentence.